microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -155,79 +135,52 @@ func reconcileResource(ctx context.Context, dynamicClient *dynamic.DynamicClient
 		if res == "" || !strings.Contains(res, "kind:") {
 			continue
 		}
-		// if create failed, wait 2s and retry, this will be removed when the monitor task is implemented
-		for {
-			createdObj, err := applyResourceToK8s(ctx, dynamicClient, ns, []byte(res))
-			if err != nil {
-				fmt.Printf("Failed to reconcile resource: %v\n", err)
-			} else {
-				fmt.Printf("Success to reconcile %s: %s\n", createdObj.GetKind(), createdObj.GetName())
+		createdObj, err := applyResourceToK8s(ctx, client, ns, svc, []byte(res))
```

Comment:
please try to combine the call to `applyResourceToK8s` with the one on Line 387  ( in this  `preProcessUserConfigmap` method ) since that call is time consuming.

Reply:
the adjust configmap is supposed be written before the manifests are applied, right?